website/docusaurus.config.js

```diff
@@ -235,7 +235,7 @@ module.exports = {
       apiKey: '47ecd3b21be71c5822571b9f59e52544',
       indexName: 'docusaurus-2',
       searchParameters: {
-        facetFilters: [`version:${versions[0]}`],
+        facetFilters: [`version:last`],
```

Comment:
This change will make the search filters only on one version. The `"last'` one.

Reply:
Actually we would like the searchParameters to be injected from the version of the page you search on.

The current behaviour is not what we would expect. For example, [searching on the page `2.0.0-alpha.58`](https://v2.docusaurus.io/docs/2.0.0-alpha.58/) filters on the latest version (alpha 62) while we would like to filter on the alpha 58.

Would it be possible to filter on the version of the page instead of only the latest?

Reply:
@s-pace this is something we discussed with @francoischalifour already but we'll do that later.

Historically we only search in the last version. This PR only focus on not breaking search on new publish, due to the facetFilter changing name.

"contextual search" is more complicated to handle. We now support multiple docs plugin instances on same site, (like ios sdk 1.0/2.0 while android sdk is 1.0/1.1/1.2 etc...) so we should figure out a contextual search experience that account for this. Will be done in another PR later.

Reply:
So that we keep track of this, opened an issue: https://github.com/facebook/docusaurus/issues/3396

Reply:
As a temporary solution you could also try the [`optionalFilters`](https://www.algolia.com/doc/api-reference/api-parameters/optionalFilters/#:~:text=Optional%20filtering%20behaves%20like%20normal,do%20not%20match%20the%20filters.) search parameter.

Reply:
Thanks, don't know much about Algolia apis :) 

But this affects scoring yet would still allow to display duplicate results no? For example if a query match only 1 doc of each version, you'll still end up with results showing the same doc in multiple versions? I mean it just "boost" the score of one specific version right?

Reply:
Hmm yes sorry, I just checked and it's not supported on the DocSearch plan anyway.